Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -37,39 +40,121 @@ namespace MultiplayerSample
     {
         m_effect = GetExplosionEffect();
         m_effect.Initialize();
+
+#if AZ_TRAIT_CLIENT
+        // The energy ball particle effect defaults to active in the component, so initialize wasActive to true for the first state.
+        m_wasActive = true;
+        AZ::TickBus::Handler::BusConnect();
+#endif
     }
 
     void EnergyBallComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
+#if AZ_TRAIT_CLIENT
+        AZ::TickBus::Handler::BusDisconnect();
+#endif
     }
 
-#if AZ_TRAIT_CLIENT
-    void EnergyBallComponent::HandleRPC_BallLaunched([[maybe_unused]] AzNetworking::IConnection* invokingConnection, [[maybe_unused]] const AZ::Vector3& location)
+    void EnergyBallComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
```

Comment:
You could actually, if you wanted, attach an event handler to the BallActive network property so that when the value changes your callback gets invoked. Since it's just an on/off thing I think it's safe to just say `if (active) startfx else killfx` although you've clearly run into some hair pulling bugs with pkfx.

Only benefit is if somebody goes crazy and adds 100,000 energy balls to a level the game doesn't have to go through and tick all 100,000 every frame on the client and the server, you'd only process effects changes when the state of BallActive changes.

Reply:
ooh, neat, i didn't know the "on changed" events existed! i'll switch over to using that.